packages/amplication-git-utils/src/git/github.service.ts

```diff
@@ -60,6 +62,121 @@ export class GithubService implements GitProvider {
       privateKey,
     });
   }
+  async getMyCommitsList(args: GetBranchArgs): Promise<Commit[]> {
+    const { branchName, owner, repositoryName } = args;
+    const botData = await this.getAmplicationBotData();
+
+    let moreCommitsPagination = true;
+    let commitsList: Commit[] = [];
+    let cursor: string | undefined = undefined;
+
+    do {
+      const { commits, hasNextPage, endCursor } =
+        await this.paginatedCommitsList({
+          botData,
+          branchName,
+          owner,
+          repositoryName,
+          cursor,
+        });
+      moreCommitsPagination = hasNextPage;
+      commitsList = commitsList.concat(commits); // The list is in ascending order ( newest commits are first )
+      cursor = endCursor;
+    } while (moreCommitsPagination);
```

Comment:
Should we add another stop here for extreme cases? (e.g. 100 pages?) 
(maybe in the future, we can think of a way to create a snapshot that includes x commits from more than X months ago, so we will not increase the number of commits infinitely)

Reply:
Its a good idea, but its a separated issue form this